bin/oref0-determine-basal.js

```diff
@@ -20,81 +20,70 @@ var determine_basal = require('oref0/lib/determine-basal/determine-basal');
 /* istanbul ignore next */
 if (!module.parent) {
     var argv = require('yargs')
-      .usage("$0 iob.json currenttemp.json glucose.json profile.json [[--auto-sens] autosens.json] [meal.json] [--reservoir reservoir.json] --currentTime [currentTime]")
+      .usage("$0 <iob.json> <currenttemp.json> <glucose.json> <profile.json> [--auto-sens <autosens.json>] [--meal <meal.json>] [--reservoir <reservoir.json>] [--currentTime <currentTime>]")
       .option('auto-sens', {
         alias: 'a',
+        nargs: 1,
         describe: "Auto-sensitivity configuration",
-        default: true
-
+        default: false
       })
       .option('currentTime', {
         alias: 'c',
+        nargs: 1,
         describe: "Override current time (for R&D only - disables safety checks)",
         default: false
-
       })
       .option('reservoir', {
         alias: 'r',
+        nargs: 1,
         describe: "Reservoir status file for SuperMicroBolus mode (oref1)",
         default: false
-
       })
       .option('meal', {
+        alias: 'm',
+        nargs: 1,
         describe: "json doc describing meals",
-        default: true
-
+        default: false
       })
       .option('missing-auto-sens-ok', {
         describe: "If auto-sens data is missing, try anyway.",
+        boolean: true,
         default: true
-
       })
       .option('missing-meal-ok', {
         describe: "If meal data is missing, try anyway.",
+        boolean: true,
         default: true
-
       })
       .option('microbolus', {
         describe: "Enable SuperMicroBolus mode (oref1)",
+        boolean: true,
         default: false
-
       })
       // error and show help if some other args given
+      .demand(4)
       .strict(true)
       .help('help')
     ;
-    function usage ( ) {
-      argv.showHelp( );
-    }
 
     var params = argv.argv;
+    var inputs = params._;
     var errors = [ ];
     var warnings = [ ];
 
-    var iob_input = params._[0];
-    if ([null, '--help', '-h', 'help'].indexOf(iob_input) > 0) {
-
-      usage( );
-      process.exit(0)
-    }
-    var currenttemp_input = params._[1];
-    var glucose_input = params._[2];
-    var profile_input = params._[3];
-    var meal_input = params._[4];
-    var autosens_input = params.autoSens;
-    if (params._.length > 5) {
-      autosens_input = params.autoSens ? params._[4] : false;
-      meal_input = params._[5];
-    }
-    if (params.meal && params.meal !== true && !meal_input) {
-      meal_input = params.meal;
+    if (inputs.length > 4) {
+      argv.showHelp( );
+      console.error('Too many arguments');
+      process.exit(1);
```

Comment:
Are you disallowing a formerly allowed arrangement of arguments here?

Reply:
Yes,
--auto-sens and --meal are now required to pass auto-sens and meal input files rather than trying to figure out whether the user entered them as non-hyphenated arguments vs. the --auto-sens <file> or --meal <file> argument method. Now there is no ambiguity that has to be resolved.

oref0-pump-loop.sh and examples.sh are modified to account for the reduced complexity on the way arguments can be provided to the script.

Reply:
So if we're going to introduce a change that breaks backwards compatibility (which I'm not opposed to on principle), we need to carefully think through the transition plan, particularly as to whether existing rigs will keep working after a `git pull` without a new run of `oref0-setup`.  Have you tested that, or can you verify that none of the remaining aliases use the old argument format, and that every call is done from within one of the scripts you updated?

Reply:
Good point on the aliases. I'll check those.

Any thoughts on how to go about making sure I find them all?

Reply:
It might be easiest to start removing the `device`s configured to use these scripts, and any reports configured to use those devices.  Then at some point we'll need to actually test on a real rig running dev that everything still works without the aliases...

Reply:
If you don't have a rig handy that you want to do a clean install on, I think we're ready to do so on at least one of ours soon.